core/src/main/java/com/yahoo/oak/common/intbuffer/OakIntBufferComparator.java

```diff
@@ -29,16 +30,47 @@ public int compareKeys(ByteBuffer buff1, ByteBuffer buff2) {
     public int compareSerializedKeys(OakScopedReadBuffer serializedKey1, OakScopedReadBuffer serializedKey2) {
         OakUnsafeDirectBuffer unsafeKey1 = (OakUnsafeDirectBuffer) serializedKey1;
         OakUnsafeDirectBuffer unsafeKey2 = (OakUnsafeDirectBuffer) serializedKey2;
-        return compare(unsafeKey1.getByteBuffer(), unsafeKey1.getOffset(), size,
-                unsafeKey2.getByteBuffer(), unsafeKey2.getOffset(), size);
+        return compare(unsafeKey1.getAddress(), size,
+                unsafeKey2.getAddress(), size);
     }
 
     @Override
     public int compareKeyAndSerializedKey(ByteBuffer key, OakScopedReadBuffer serializedKey) {
         OakUnsafeDirectBuffer unsafeKey = (OakUnsafeDirectBuffer) serializedKey;
-        return compare(key, 0, size, unsafeKey.getByteBuffer(), unsafeKey.getOffset(), size);
+        return compare(key, 0, size, unsafeKey.getAddress(), size);
     }
 
+    
+    public static int compare(long buff1, int size1, long buff2, int size2) {
```

Comment:
Also here, I think, we can avoid using both `OakUnsafeDirectBuffer` and `UnsafeUtils` and stick with given `OakScopedReadBuffer` interface...